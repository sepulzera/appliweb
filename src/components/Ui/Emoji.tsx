import * as React from 'react';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  emoji: {
    fontFamily: [
      'apple color emoji',
      'segoe ui emoji',
      'noto color emoji',
      'android emoji',
      'emojisymbols',
      'emojione mozilla',
      'twemoji mozilla',
      'segoe ui symbol',
      'sans-serif',
    ].join(','),
  },
});

/** {@link Emoji} Props. */
export interface IEmojiProps {
  /** Alternative text used by assistive technologies. If not set, this emoji won't be visible to screen readers. */
  label?:     string;
  /** Emoji. */
  children:   React.ReactNode;
}

/**
 * Renders an accessible unicode emoji.
 *
 * @param props - {@link IEmojiProps}.
 */
const Emoji: React.FC<IEmojiProps> = (props: IEmojiProps) => {
  const classes = useStyles();
  return (
    <span className={classes.emoji} role='img' aria-label={props.label || ''}>{props.children}</span>
  );
};

export default Emoji;
