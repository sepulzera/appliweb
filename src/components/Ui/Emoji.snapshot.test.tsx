/* eslint-disable jsx-a11y/accessible-emoji */
import React from 'react';
import renderer from 'react-test-renderer';

import Emoji from './Emoji';

test('render emoji with no label', () => {
  const tree = renderer.create(<Emoji>ğŸ‘‹</Emoji>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render emoji with label', () => {
  const tree = renderer.create(<Emoji label='Waving hand'>ğŸ‘‹</Emoji>).toJSON();
  expect(tree).toMatchSnapshot();
});
