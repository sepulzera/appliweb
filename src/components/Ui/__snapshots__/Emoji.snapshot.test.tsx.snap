// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render emoji with label 1`] = `
<span
  aria-label="Waving hand"
  className="makeStyles-emoji-1"
  role="img"
>
  ðŸ‘‹
</span>
`;

exports[`render emoji with no label 1`] = `
<span
  aria-label=""
  className="makeStyles-emoji-1"
  role="img"
>
  ðŸ‘‹
</span>
`;
